### PR TITLE
[ISSUE #1057]Filter out docker0 when finding a local address

### PR DESCRIPTION
**Pull request by dut-xiayuan**
## What is the purpose of the change

To filter out docker0 when finding a local address

If the address of docker0 bridge was set as brokerIP1 or brokerIP2 of the master node, the client or slave couldn't establish a connection with it.

## Brief changelog

Filter out docker0 when finding a local address

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingUtil.java [hunk 1]
```diff
@@ -99,6 +99,10 @@ public static String getLocalAddress() {
             ArrayList<String> ipv6Result = new ArrayList<String>();
             while (enumeration.hasMoreElements()) {
                 final NetworkInterface networkInterface = enumeration.nextElement();
+                if ("docker0".equals(networkInterface.getName())) {
```
Comment (vongosling) [inline remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingUtil.java]: docker1 could also exist? docker bridge is a configuration option in docker architecture. could we have a common way to sweep out these ips?

Comment (dut-xiayuan) [reply remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingUtil.java]: Hi, we can read the sysfs directory to check if the interface is a bridge. This method is also used in some bridge tools, such as `brctl`, in fact, I was inspired by this tool.

Comment (dut-xiayuan) [reply remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingUtil.java]: more explantion:
http://man7.org/linux/man-pages/man5/sysfs.5.html
https://unix.stackexchange.com/questions/40560/how-to-know-if-a-network-interface-is-tap-tun-bridge-or-physical
https://github.com/Gandi/bridge-utils/blob/master/libbridge/libbridge_init.c

**Comment by ShannonDing** [comment]
could you pls move this PR to develop branch?

**Comment by dut-xiayuan** [comment]
ok, reset commit branch to develop

**Pull request merged**